src/community/json-ld/src/main/java/org/geoserver/jsonld/request/JsonLdTemplateCallBackOGC.java

```diff
@@ -0,0 +1,90 @@
+ * application directory.
+ */
+package org.geoserver.jsonld.request;
+
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.config.GeoServer;
+import org.geoserver.jsonld.builders.impl.RootBuilder;
+import org.geoserver.jsonld.configuration.JsonLdConfiguration;
+import org.geoserver.jsonld.expressions.ExpressionsUtils;
+import org.geoserver.jsonld.response.JSONLDGetFeatureResponse;
+import org.geoserver.ows.AbstractDispatcherCallback;
+import org.geoserver.ows.DispatcherCallback;
+import org.geoserver.ows.Request;
+import org.geoserver.platform.Operation;
+import org.geoserver.platform.ServiceException;
+import org.geoserver.util.XCQL;
+import org.geotools.filter.text.ecql.ECQL;
+import org.opengis.filter.Filter;
+
+/**
+ * This {@link DispatcherCallback} implementation OGCAPI compliant that checks on operation
+ * dispatched event if a json-ld path has been provided to cql_filter and evaluate it against the
+ * {@link org.geoserver.jsonld.builders.JsonBuilder} tree to get the corresponding {@link Filter}
+ */
+public class JsonLdTemplateCallBackOGC extends AbstractDispatcherCallback {
+
+    private Catalog catalog;
+
+    private JsonLdConfiguration configuration;
+
+    public JsonLdTemplateCallBackOGC(GeoServer gs, JsonLdConfiguration configuration) {
+        this.catalog = gs.getCatalog();
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Operation operationDispatched(Request request, Operation operation) {
+        boolean isJsonLd =
+                request.getKvp().get("f") != null
+                                && request.getKvp().get("f").equals(JSONLDGetFeatureResponse.MIME)
+                        ? true
+                        : false;
+        if (request.getHttpRequest().getServletPath().equals("/ogc") && isJsonLd) {
+            try {
+                String strFilter = (String) request.getKvp().get("FILTER");
+                if (strFilter != null && strFilter.indexOf(".") != -1) {
+                    FeatureTypeInfo typeInfo =
+                            getFeatureType((String) operation.getParameters()[0]);
+                    RootBuilder root = configuration.getTemplate(typeInfo);
+                    if (root != null) {
+                        JsonLdPathVisitor visitor =
+                                new JsonLdPathVisitor(typeInfo.getFeatureType());
+                        /* Todo find a better way to replace json-ld path with corresponding template attribute*/
+                        // Get filter from string in order to make it accept the visitor
+                        Filter f = (Filter) XCQL.toFilter(strFilter).accept(visitor, root);
+                        // Taking back a string from Function cause
+                        // OGC API get a string cql filter from query string
+                        String newFilter = ECQL.toCQL(f).replaceAll("\"", "").replaceAll("/", ".");
```

Comment:
Hmm... unsure about these replacement, some names have to be quoted, say for example an attribute is called "between", just like one of the CQL keywords..

Reply:
now it's checking if after "\"" there is a space char assuming thus that an eventual xpath ended. Do you think this would be enough?